tests/queries/0_stateless/01941_multiple_pipe_read.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+SAMPLE_FILE="$CURDIR/01941_sample_data.csv"
+STD_ERROR_CAPTURED="$CURDIR/01941_std_error_captured.log"
+
+echo 'File generated:'
+${CLICKHOUSE_LOCAL} -q "SELECT number, if(number in (4,6), 'AAA', 'BBB') from numbers(7) FORMAT TSV" | tr '\t' ',' >"$SAMPLE_FILE"
+cat "$SAMPLE_FILE"
+
+echo '******************'
+echo 'Attempt to read twice from a pipeline'
+cat ${CLICKHOUSE_LOCAL} --structure 'key String' -q 'select * from table; select * from table;' <<<foo 2>"$STD_ERROR_CAPTURED"
+ret = $?
+echo "Return code: $ret"
```

Comment:
you can use `set -e` instead of manually checking return code every time (`set -e` is not perfect, but will work for you)